src/lib.rs

```diff
@@ -299,24 +299,50 @@ where
 {
     /// Creates a new, empty map with the default initial table size (16).
     pub fn new() -> Self {
+        Self::with_hasher(RandomState::new())
+    }
+
+    /// Creates a new, empty map with an initial table size accommodating the specified number of
+    /// elements without the need to dynamically resize.
+    pub fn with_capacity(n: usize) -> Self {
+        Self::with_capacity_and_hasher(RandomState::new(), n)
+    }
+}
+
+impl<K, V, S: BuildHasher> FlurryHashMap<K, V, S> {
+    /// Creates an empty HashMap which will use the given hash builder to hash keys.
+    ///
+    /// The created map has the default initial capacity.
+    ///
+    /// Warning: hash_builder is normally randomly generated, and is designed to
+    /// allow HashMaps to be resistant to attacks that cause many collisions and
+    /// very poor performance. Setting it manually using this
+    /// function can expose a DoS attack vector.
+    pub fn with_hasher(hash_builder: S) -> Self {
         Self {
             table: Atomic::null(),
             next_table: Atomic::null(),
             transfer_index: AtomicIsize::new(0),
             count: AtomicUsize::new(0),
             size_ctl: AtomicIsize::new(0),
-            build_hasher: RandomState::new(),
+            build_hasher: hash_builder,
         }
     }
 
-    /// Creates a new, empty map with an initial table size accommodating the specified number of
-    /// elements without the need to dynamically resize.
-    pub fn with_capacity(n: usize) -> Self {
+    /// Creates an empty HashMap with the specified capacity, using hash_builder to hash the keys.
+    ///
+    /// The hash map will be able to hold at least capacity elements without reallocating. If
+    /// capacity is 0, the hash map will not allocate.
```

Comment:
I'm not sure we actually give the _guarantee_ that it will not have to reallocate. If, for example, all the keys has to the same bin, we will resize sooner. It's certainly a degenerate case, but we don't want to promise something we cannot guarantee. How about:

```rust
// The map will be sized to accommodate `capacity` elements with a low chance of reallocating (assuming uniformly distributed hashes). If `capacity` is 0, the call will not allocate, and is equivalent to [`FlurryHashMap::new`].
```